### PR TITLE
Adds a notification in dchat when a player dies

### PR DESCRIPTION
**Pull request by SabreML**
## What Does This PR Do

Shows a message to observers whenever someone dies, showing their name and location as well as a `(JMP)` jump button.
This is enabled by default, but can be disabled with the `Show/Hide Death Notifications` setting in the Preferences tab.

Also fixed a small typo in the `Show/Hide GhostPDA` setting.

## Why It's Good For The Game

Provides a more interesting experience for observers.
*(Probably more when I'm not so tired)*

If someone doesn't want the messages for any reason, then there's a preferences toggle to disable it.

## Images of changes

death message

## Changelog
:cl:
add: A player dying will now display a message in dchat with their name and location. (Enabled by default, this can be disabled with 'Show/Hide Death Notifications')
spellcheck: Removed a '.' in the 'Ghost PDA' setting toggle
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
SQL Query to be ran when merged to auto-enable this for all players
```sql
UPDATE `player` SET `toggles_2` = (`toggles_2` + 128) WHERE (`toggles_2` & 128) < 1;
```

File: code/modules/client/preference/preferences_toggles.dm [hunk 1]
```diff
@@ -293,7 +293,7 @@
 /client/verb/toggle_ghost_pda()
 	set name = "Show/Hide GhostPDA"
 	set category = "Preferences"
-	set desc = ".Toggle seeing PDA messages as an observer."
+	set desc = "Toggle seeing PDA messages as an observer."
```
Comment (farie82) [inline code/modules/client/preference/preferences_toggles.dm:296]: Undocumented change.

**Review by farie82 — APPROVED** [review]
LGTM. Be sure to add the commented on change in your CL and PR description as well

File: code/modules/client/preference/preferences_toggles.dm (hunk 1 above)
Comment (SabreML) [reply code/modules/client/preference/preferences_toggles.dm:296]: Wasn't sure if I needed to mention that since it was just fixing a typo, but why not.

**Review by Kyep — APPROVED** [review]
Approved, though I wish for admins it listed the last person to attack them, and whether their last attacker was an antag or not.

File: code/modules/mob/living/death.dm [hunk 2]
```diff
@@ -91,6 +91,14 @@
 	GLOB.alive_mob_list -= src
 	GLOB.dead_mob_list += src
 	if(mind)
+		if(mind.name && !isbrain(src)) // !isbrain() is to stop it from being called twice
+			var/turf/T = get_turf(src)
+			var/area_name = get_area_name(T)
+			for(var/P in GLOB.player_list)
```
Comment (AffectedArc07) [inline code/modules/mob/living/death.dm:97]: `GLOB.dead_mob_list` may work here and will be a smaller list to iterate over.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by SabreML** [comment]
> Approved, though I wish for admins it listed the last person to attack them, and whether their last attacker was an antag or not.

It should do that anyway in the admin logs, this is just for everyone else.

File: code/modules/mob/living/death.dm (hunk 2 above)
Comment (farie82) [reply code/modules/mob/living/death.dm:97]: That will also include dead monkeys. Dead simple mobs and what not. You then also need to check if the mob has a client which is "free" in this list

Comment (SabreML) [reply code/modules/mob/living/death.dm:97]: If it counts all dead mobs it'll probably be better to just keep using `GLOB.player_list`.

Comment (AffectedArc07) [reply code/modules/mob/living/death.dm:97]: All observers are put into that list, and its used in other places too.
```dm
		// Show it to ghosts
		for(var/mob/M in GLOB.dead_mob_list)
			if(isobserver(M) && M.client && (M.client.prefs.toggles & PREFTOGGLE_CHAT_GHOSTPDA))
				var/ghost_message = "<span class='name'>[pda.owner]</span> ([ghost_follow_link(pda, ghost=M)]) <span class='game say'>PDA Message</span> --> <span class='name'>[P.owner]</span> ([ghost_follow_link(P, ghost=M)]): <span class='message'>[t]</span>"
				to_chat(M, "[ghost_message]")
```
Please change it

**Review by AffectedArc07 — APPROVED** [review]

**Pull request merged**